src/modules/esl-avatar/core/esl-avatar.ts

```diff
@@ -0,0 +1,109 @@
+import {ExportNs} from '../../esl-utils/environment/export-ns';
+import {ESLBaseElement} from '../../esl-base-element/core';
+import {attr, bind, boolAttr, prop} from '../../esl-utils/decorators';
+
+@ExportNs('Avatar')
+export class ESLAvatar extends ESLBaseElement {
+  public static override is = 'esl-avatar';
+  public static observedAttributes = ['image-src', 'username'];
+
+  /** Event to dispatch on change of {@link ESLAvatar} */
+  @prop('esl:avatar:changed') public AVATAR_CHANGED_EVENT: string;
+
+  /** Source path of the avatar image */
+  @attr() public imageSrc: string;
+  /** The limit number of letters to be displayed in text-only mode */
+  @attr({defaultValue: 2}) public limit: number;
+  /** Policy of loading image that is outside of the viewport */
+  @attr({defaultValue: 'lazy'}) public loading: 'eager' | 'lazy';
+  /** The name of the user for whom the avatar is displayed */
+  @attr({defaultValue: ''}) public username: string;
+
+  /** @readonly Marker of displaying mode with image */
+  @boolAttr({readonly: true}) public withImg: boolean;
+  /** @readonly Marker of displaying mode without image, only text */
+  @boolAttr({readonly: true}) public textOnly: boolean;
```

Comment:
It looks like that's the same or 'concurrent' property:
so we can use one boolen marker attribute 
or string (enumed) attribute in case we have more then 2 states (e.g. empty)